lib/puppet/functions/vault_lookup/lookup.rb

```diff
@@ -1,10 +1,26 @@
 Puppet::Functions.create_function(:'vault_lookup::lookup') do
-  dispatch :lookup do
+  dispatch :lookup_without_key do
     param 'String', :path
-    optional_param 'String', :vault_url
+    optional_param 'Stdlib::HTTPUrl', :vault_url
```

Comment:
The first matching dispatch should 'win'.  So if there are 2 parameters and the second one *isn't* a url, it should match the second dispatch and be treated as the key.